### ISSUE TITLE
Support for encrypted file ediff(ing)

### ISSUE THREAD

**Issue by sochotnicky** [tuh8888/chezmoi.el#22]
Thanks for the package. It makes managing chezmoi dotfiles much easier.

One thing that is a bit tricky is dealing with files that are encrypted. I imagine this might be ideally split off to a partially separate package to support age encrypted files. 

Then I suppose what chezmoi.el could do is:
 1. Decrypt the chezmoi file into a temporary directory with appropriate permissions
 2. Diff/merge the target file and the temporarily decrypted file
 3. If the decrypted file changed - reencrypt once again.

This is a relatively minor thing since 99.9% of my files are not encrypted, but once in a while it would come in very useful.

**Comment by sochotnicky** [comment]
I haven't found any packages with support for age encryption in emacs so I'll work on that most likely. After that it might not be too much work to add support for chezmoi (it would require loading identities/recipients from config file but that should not be too difficult). I might contribute a PR for that if you're up for it.

**Comment by tuh8888** [comment]
If you can put together a PR that at least gets the encrypted file into a buffer, I can take a look at how we might wrap the process with chezmoi state management. I don't use encrypted files myself (mainly I use Bitwarden to inject values when needed), but I'm sure this could be useful if you do

**Comment by sochotnicky** [comment]
Age encryption has some use elsewhere so I'd probably create a separate package. It will be a while but we can keep this issue open and I'll update once I have some news?

**Comment by sochotnicky** [comment]
So I 've actually put up an experimental version of the package to support age decryption here:
https://codeberg.org/sochotnicky/age-mode

I've actually tested the above with following change in chezmoi:
https://gist.github.com/sochotnicky/70cdcef68ca693b3beebcde8546f030e

And it actually allows nice merging of encrypted files.

The biggest change is that chezmoi needs to trigger ediff on buffers and not files (for some reason ediff on files doesn't trigger proper file handlers). And chezmoi needs to set a buffer-local variable to set recipients from the config file properly so that files get encrypted to the right folks. That said above patch is incomplete but the switch to ediff-buffers would help folks who use GPG encrypted files too. So it might be worth doing on its own. 

Nice thing is that given how file handlers work - chezmoi does not need to depend on age directly as long as it sets the buffer local variable correctly. If folks have the above library installed it would just work.

**Comment by tuh8888** [comment]
Sorry @sochotnicky it took me a little while to get around to checking out your patch. I noticed that you've deprecated your package in favor of `age.el`. That package seems to handle age recipients and identities itself. I've created a PR (#24) that should gently help when ediffing age files. I stuck with the `ediff-files` approach as it appears to work, but let me know if it's still not enough.

One important note is that I've added the ability to "--force" write which is important for making sure that these encrypted files look the same to Chezmoi when their contents are identical.
